beacon.go

```diff
@@ -71,58 +68,30 @@ func query(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	refName, allele, coord, err := parseInput(r)
+	query, err := parseInput(r)
 	if err != nil {
+		http.Error(w, fmt.Sprintf("parsing input: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	if err := query.ValidateInput(); err != nil {
 		http.Error(w, fmt.Sprintf("validating input: %v", err), http.StatusBadRequest)
 		return
 	}
 
 	ctx := appengine.NewContext(r)
-	exists, err := genomeExists(ctx, refName, allele, coord)
+	exists, err := query.Execute(ctx, config.ProjectID, config.TableID)
 	if err != nil {
 		http.Error(w, fmt.Sprintf("computing result: %v", err), http.StatusInternalServerError)
 		return
 	}
 
 	if err := writeResponse(w, exists); err != nil {
-		http.Error(w, fmt.Sprintf("validating server configuration: %v", err), http.StatusInternalServerError)
+		http.Error(w, fmt.Sprintf("writing response: %v", err), http.StatusInternalServerError)
```

Comment:
Sorry I didn't catch this in the previous review, but this probably won't work: by the time an error has occurred in the writeResponse function you've already started writing an HTTP response and this will fail (as soon as you call WriteHeader you can't change the response code).